.yamato/scripts/update_from_upstream.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh
+
+# Prequisites:
+# This script assumes git and github CLR (gh) are installed.
+
+# Cause the script to fail if any individual command fails
+set -e
+
+# Fetch from upstream and to a fast forward merge into our local main
+git remote add upstream https://github.com/dotnet/runtime
+git fetch upstream main
+git merge --ff-only upstream/main
```

Comment:
@joncham It looks like a fast forward merge is failing with the error:

```
fatal: Not possible to fast-forward, aborting.
```

So maybe we should keep this as a rebase. That way our `main` always has the same history as upstream's `main`.

Reply:
If we cannot fast forward that indicates a problem AFAIK. It means either we commited to `main` (wrong) or upstream rewrote history?

Reply:
Ahh, you are correct. I was not checking out `main`, the clone was still on `unity-main`, which is why this failed. With the latest code the fast forward merge works.